flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -259,6 +264,12 @@ public SingleInputGate(
         this.unpooledSegment = MemorySegmentFactory.allocateUnpooledSegment(segmentSize);
         this.bufferDebloater = bufferDebloater;
         this.throughputCalculator = checkNotNull(throughputCalculator);
+
+        this.tieredStorageConsumerClient =
+                tieredStorageConfiguration.enabledTieredStorage()
+                        ? new TieredStorageConsumerClient(
+                                tieredStorageConfiguration.getTierFactories())
+                        : null;
```

Comment:
Should create this in `SingleInputGateFactory`.

Reply:
Done.